kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/DataMatchDataConsistencyCalculateAlgorithm.java

```diff
@@ -154,9 +158,18 @@ private CalculationContext createCalculationContext(final DataConsistencyCalcula
     
     private void fulfillCalculationContext(final CalculationContext calculationContext, final DataConsistencyCalculateParameter param) throws SQLException {
         String sql = getQuerySQL(param);
-        PreparedStatement preparedStatement = setCurrentStatement(calculationContext.getConnection().prepareStatement(sql));
+        DatabaseType databaseType = TypedSPILoader.getService(DatabaseType.class, param.getDatabaseType());
+        PreparedStatement preparedStatement;
+        if (databaseType instanceof MySQLDatabaseType) {
+            preparedStatement = setCurrentStatement(JDBCStreamQueryUtil.generateMySQLStreamQueryPreparedStatement(calculationContext.getConnection(), sql));
+        } else if (databaseType instanceof PostgreSQLDatabaseType || databaseType instanceof OpenGaussDatabaseType) {
+            preparedStatement = setCurrentStatement(JDBCStreamQueryUtil.generatePostgreSQLStreamQueryPreparedStatement(calculationContext.getConnection(), sql, chunkSize));
+        } else {
+            log.warn("not support {} streaming query now, pay attention to memory usage", databaseType.getType());
+            preparedStatement = setCurrentStatement(calculationContext.getConnection().prepareStatement(sql));
+            preparedStatement.setFetchSize(chunkSize);
+        }
```

Comment:
Could we extract this code block into `JDBCStreamQueryUtil`? Since there's the same code block in `InventoryDumper`